### PR TITLE
[CAS-15] Expose deposit instructions required for Realms integration

### PR DESCRIPTION
**Pull request by adamlawrencium**

### CODE REVIEW

File: sdk/src/client.ts [hunk 1]
```diff
@@ -48,7 +48,7 @@ export class VaultClient {
     private constructor(
         public program: anchor.Program<CastleLendingAggregator>,
         public vaultId: PublicKey,
-        private vaultState: Vault,
+        public vaultState: Vault,
```
Comment (charlieyou) [inline sdk/src/client.ts:51]: Add getters instead pls

File: sdk/src/client.ts [hunk 2]
```diff
@@ -965,6 +972,70 @@ export class VaultClient {
...
 
+    /**
+     * Creates LP ATA creation and deposit ix transactions for Realms (https://realms.today/)
+     * @param reserveTokenOwner DAO's wallet
+     * @param amount amount
+     * @param userReserveTokenAccount DAO's reserve token account
+     * @param lpTokenAccountFeePayer The proposal creator that pays for the LP ATA creation
+     * @returns
+     */
+    async realmsDepositIxs({
```
Comment (charlieyou) [inline sdk/src/client.ts:983]: If this is for realms only, why have it in the sdk?

File: sdk/src/client.ts [hunk 3]
```diff
@@ -1042,17 +1113,18 @@ export class VaultClient {
 }
 
 const createAta = (
-    wallet: anchor.Wallet,
+    pubkey: PublicKey,
```
Comment (charlieyou) [inline sdk/src/client.ts:1116]: Can you rename to `owner` or something more descriptive?

**Review by charlieyou — CHANGES_REQUESTED** [review]
Also: patch version bump

File: sdk/src/client.ts (hunk 2 above)
Comment (adamlawrencium) [reply sdk/src/client.ts]: True, no need for this specific of a method inside the class

File: sdk/src/client.ts (hunk 3 above)
Comment (adamlawrencium) [reply sdk/src/client.ts]: Done

**Review by charlieyou — APPROVED** [review]

**Pull request merged**